[Automation] java run fails for software.amazon.awssdk:s3:2.30.0

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24810004413/job/72612756770
Last 300 lines of the log:
```console

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 700ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: software.amazon.awssdk:s3:2.25.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :jar
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

6 tests completed, 1 failed

> Task :test FAILED


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 3 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:s3:2.25.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 6s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.30.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 749ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 965ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 721ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 2 executed, 8 up-to-date

6 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 3s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.30.0 failed with exit code 1.
FAILED[java run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24810004413/job/72612756787
Last 300 lines of the log:
```console

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 747ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: software.amazon.awssdk:s3:2.25.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :jar
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

6 tests completed, 1 failed

> Task :test FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> 
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 3 executed, 8 up-to-date
Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:s3:2.25.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 6s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.30.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 737ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 996ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 841ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava

Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

6 tests completed, 1 failed

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
10 actionable tasks: 2 executed, 8 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.30.0 failed with exit code 1.
FAILED[java run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24810004413/job/72612756819
Last 300 lines of the log:
```console

BUILD SUCCESSFUL in 830ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: software.amazon.awssdk:s3:2.25.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :jar
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

6 tests completed, 1 failed

> Task :test FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 3 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:s3:2.25.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.30.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.30.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 804ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 software.amazon.awssdk:s3:2.25.3:2.30.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava

> Task :processTestResources NO-SOURCE
> Task :testClasses
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/src/test/java/software_amazon_awssdk/s3/S3Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :test

S3Test > deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() FAILED
    java.lang.AssertionError: 
    Expecting Optional to contain:
      "L+s059sh7zk0NtQGtBsdXQ=="
    but was empty.
        at software_amazon_awssdk.s3.S3Test.deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults(S3Test.java:340)

6 tests completed, 1 failed


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/tests/test/index.html

* Try:
> Run with > Task :test FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/s3/2.25.3/build/reports/problems/problems-report.html
--scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 2 executed, 8 up-to-date

BUILD FAILED in 3s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.30.0 failed with exit code 1.
FAILED[java run][2.30.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.30.0" ./gradlew clean javaTest -Pcoordinates="software.amazon.awssdk:s3:2.25.3"]
```

--- a/metadata/software.amazon.awssdk/s3/2.30.0/reachability-metadata.json
+++ b/metadata/software.amazon.awssdk/s3/2.30.0/reachability-metadata.json
@@ -1,0 +1,84 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3Client"
+      },
+      "type": "com.sun.crypto.provider.HmacCore$HmacSHA256",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3Client"
+      },
+      "type": "com.sun.xml.internal.stream.XMLInputFactoryImpl",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3Client"
+      },
+      "type": "sun.security.provider.SHA2$SHA256",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3Client"
+      },
+      "glob": "META-INF/services/java.net.spi.URLStreamHandlerProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.auth.scheme.internal.DefaultS3AuthSchemeProvider"
+      },
+      "glob": "META-INF/services/java.net.spi.URLStreamHandlerProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3Client"
+      },
+      "glob": "META-INF/services/javax.xml.stream.XMLInputFactory"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.S3Client"
+      },
+      "glob": "org/slf4j/impl/StaticLoggerBinder.class"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3ClientBuilder"
+      },
+      "glob": "software/amazon/awssdk/global/handlers/execution.interceptors"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.endpoints.internal.DefaultPartitionDataProvider"
+      },
+      "glob": "software/amazon/awssdk/global/partitions.json"
+    },
+    {
+      "condition": {
+        "typeReached": "software.amazon.awssdk.services.s3.DefaultS3BaseClientBuilder"
+      },
+      "glob": "software/amazon/awssdk/services/s3/execution.interceptors"
+    }
+  ]
+}

--- a/metadata/software.amazon.awssdk/s3/index.json
+++ b/metadata/software.amazon.awssdk/s3/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.30.0",
+    "tested-versions" : [
+      "2.30.0"
+    ],
+    "allowed-packages" : [
+      "software.amazon.awssdk"
+    ]
+  },
+  {
     "metadata-version" : "2.25.3",
     "source-code-url" : "https://repo1.maven.org/maven2/software/amazon/awssdk/s3/$version$/s3-$version$-sources.jar",
     "repository-url" : "https://github.com/aws/aws-sdk-java-v2",

--- a/metadata/software.amazon.awssdk/s3/index.json
+++ b/metadata/software.amazon.awssdk/s3/index.json
@@ -2,11 +2,16 @@
   {
     "latest" : true,
     "metadata-version" : "2.30.0",
+    "source-code-url" : "https://repo1.maven.org/maven2/software/amazon/awssdk/s3/$version$/s3-$version$-sources.jar",
+    "repository-url" : "https://github.com/aws/aws-sdk-java-v2",
+    "test-code-url" : "https://github.com/aws/aws-sdk-java-v2/tree/$version$/services/s3/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/software/amazon/awssdk/s3/$version$/s3-$version$-javadoc.jar",
+    "description" : "This module provides the AWS SDK for Java v2 client classes for Amazon S3. It lets Java applications interact with Amazon Simple Storage Service to manage buckets and objects.",
     "tested-versions" : [
       "2.30.0"
     ],
     "allowed-packages" : [
-      "software.amazon.awssdk"
+      "software.amazon.awssdk.services.s3"
     ]
   },
   {

--- a/stats/software.amazon.awssdk/s3/2.30.0/execution-metrics.json
+++ b/stats/software.amazon.awssdk/s3/2.30.0/execution-metrics.json
@@ -1,0 +1,93 @@
+{
+  "fix_java_run_fail:2026-05-05": {
+    "timestamp": "2026-05-05T03:31:23.769993Z",
+    "library": "software.amazon.awssdk:s3:2.30.0",
+    "starting_commit": "75c1273ea62d3e69d6d5353e56416e72de8cfb1b",
+    "ending_commit": "ed42d2a439408704447a87988e016e4dbb3f94db",
+    "previous_library": "software.amazon.awssdk:s3:2.25.3",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.30.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 28514,
+          "missed": 232606,
+          "ratio": 0.109199,
+          "total": 261120
+        },
+        "line": {
+          "covered": 7139,
+          "missed": 53774,
+          "ratio": 0.1172,
+          "total": 60913
+        },
+        "method": {
+          "covered": 1405,
+          "missed": 18034,
+          "ratio": 0.072277,
+          "total": 19439
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.25.3",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 24806,
+          "missed": 211591,
+          "ratio": 0.104934,
+          "total": 236397
+        },
+        "line": {
+          "covered": 6306,
+          "missed": 48357,
+          "ratio": 0.115361,
+          "total": 54663
+        },
+        "method": {
+          "covered": 1331,
+          "missed": 16199,
+          "ratio": 0.075927,
+          "total": 17530
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 60784,
+      "cached_input_tokens_used": 923648,
+      "output_tokens_used": 5172,
+      "iterations": 1,
+      "cost_usd": 0.617,
+      "tested_library_loc": 7139,
+      "metadata_entries": 13,
+      "previous_library_metadata_entries": 18,
+      "code_coverage_percent": 11.72,
+      "previous_library_coverage_percent": 11.54
+    },
+    "artifacts": {
+      "test_file": "tests/src/software.amazon.awssdk/s3/2.30.0/src/test/java/software_amazon_awssdk/s3/S3Test.java",
+      "metadata_file": "metadata/software.amazon.awssdk/s3/2.30.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/software.amazon.awssdk/s3/2.30.0/stats.json
+++ b/stats/software.amazon.awssdk/s3/2.30.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.30.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 28514,
+        "missed" : 232606,
+        "ratio" : 0.109199,
+        "total" : 261120
+      },
+      "line" : {
+        "covered" : 7139,
+        "missed" : 53774,
+        "ratio" : 0.1172,
+        "total" : 60913
+      },
+      "method" : {
+        "covered" : 1405,
+        "missed" : 18034,
+        "ratio" : 0.072277,
+        "total" : 19439
+      }
+    }
+  } ]
+}

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/.gitignore
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/build.gradle
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "software.amazon.awssdk:s3:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/gradle.properties
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = software.amazon.awssdk:s3:2.30.0
+library.version = 2.30.0
+metadata.dir = software.amazon.awssdk/s3/2.30.0/

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/settings.gradle
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'software.amazon.awssdk.s3_tests'

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/src/test/java/software_amazon_awssdk/s3/S3Test.java
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/src/test/java/software_amazon_awssdk/s3/S3Test.java
@@ -1,0 +1,630 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package software_amazon_awssdk.s3;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Deque;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.AbortableInputStream;
+import software.amazon.awssdk.http.ExecutableHttpRequest;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import software.amazon.awssdk.services.s3.S3Uri;
+import software.amazon.awssdk.services.s3.model.ChecksumMode;
+import software.amazon.awssdk.services.s3.model.CommonPrefix;
+import software.amazon.awssdk.services.s3.model.CompletedMultipartUpload;
+import software.amazon.awssdk.services.s3.model.CompletedPart;
+import software.amazon.awssdk.services.s3.model.CopyObjectResponse;
+import software.amazon.awssdk.services.s3.model.CreateMultipartUploadResponse;
+import software.amazon.awssdk.services.s3.model.Delete;
+import software.amazon.awssdk.services.s3.model.DeleteObjectsResponse;
+import software.amazon.awssdk.services.s3.model.DeletedObject;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.GetUrlRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.MetadataDirective;
+import software.amazon.awssdk.services.s3.model.ObjectIdentifier;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Error;
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.services.s3.model.UploadPartResponse;
+
+class S3Test {
+
+    @Test
+    void configurationAndUtilitiesSupportPathStyleUrlsAndUriParsing() {
+        S3Configuration configuration = S3Configuration.builder()
+            .pathStyleAccessEnabled(true)
+            .chunkedEncodingEnabled(false)
+            .checksumValidationEnabled(false)
+            .multiRegionEnabled(true)
+            .build();
+
+        S3Configuration copiedConfiguration = configuration.toBuilder()
+            .multiRegionEnabled(false)
+            .build();
+
+        assertThat(configuration.pathStyleAccessEnabled()).isTrue();
+        assertThat(configuration.chunkedEncodingEnabled()).isFalse();
+        assertThat(configuration.checksumValidationEnabled()).isFalse();
+        assertThat(configuration.multiRegionEnabled()).isTrue();
+        assertThat(copiedConfiguration.multiRegionEnabled()).isFalse();
+
+        RecordingHttpClient httpClient = new RecordingHttpClient();
+        try (S3Client client = createClient(
+            httpClient,
+            URI.create("https://s3.us-west-2.amazonaws.com"),
+            builder -> builder.pathStyleAccessEnabled(true)
+        )) {
+            URL pathStyleUrl = client.utilities().getUrl(GetUrlRequest.builder()
+                .bucket("docs-bucket")
+                .key("reports/quarterly summary.txt")
+                .build());
+
+            assertThat(pathStyleUrl.toString())
+                .isEqualTo("https://s3.us-west-2.amazonaws.com/docs-bucket/reports/quarterly%20summary.txt");
+
+            S3Uri parsedPathStyleUri = client.utilities().parseUri(URI.create(
+                "https://s3.us-west-2.amazonaws.com/docs-bucket/reports/quarterly%20summary.txt?versionId=v1&partNumber=7"));
+
+            assertThat(parsedPathStyleUri.bucket()).hasValue("docs-bucket");
+            assertThat(parsedPathStyleUri.key()).hasValue("reports/quarterly summary.txt");
+            assertThat(parsedPathStyleUri.region()).hasValue(Region.US_WEST_2);
+            assertThat(parsedPathStyleUri.isPathStyle()).isTrue();
+            assertThat(parsedPathStyleUri.firstMatchingRawQueryParameter("versionId")).hasValue("v1");
+            assertThat(parsedPathStyleUri.firstMatchingRawQueryParameter("partNumber")).hasValue("7");
+
+            S3Uri parsedVirtualHostUri = client.utilities().parseUri(URI.create(
+                "https://docs-bucket.s3.us-west-2.amazonaws.com/reports/archive.txt?x-id=GetObject"));
+
+            assertThat(parsedVirtualHostUri.bucket()).hasValue("docs-bucket");
+            assertThat(parsedVirtualHostUri.key()).hasValue("reports/archive.txt");
+            assertThat(parsedVirtualHostUri.region()).hasValue(Region.US_WEST_2);
+            assertThat(parsedVirtualHostUri.isPathStyle()).isFalse();
+            assertThat(parsedVirtualHostUri.firstMatchingRawQueryParameter("x-id")).hasValue("GetObject");
+        }
+    }
+
+    @Test
+    void putObjectAndGetObjectAsBytesUseExpectedMarshallingAndParsing() {
+        RecordingHttpClient httpClient = new RecordingHttpClient()
+            .enqueue(MockHttpResponse.of(
+                200,
+                Map.of(
+                    "ETag", List.of("\"put-etag\""),
+                    "x-amz-version-id", List.of("put-version")
+                ),
+                new byte[0]
+            ))
+            .enqueue(MockHttpResponse.of(
+                200,
+                Map.of(
+                    "Content-Type", List.of("text/plain"),
+                    "Content-Length", List.of("18"),
+                    "ETag", List.of("\"get-etag\""),
+                    "x-amz-meta-origin", List.of("integration-test"),
+                    "x-amz-version-id", List.of("get-version")
+                ),
+                "hello from aws sdk".getBytes(StandardCharsets.UTF_8)
+            ));
+
+        try (S3Client client = createClient(httpClient, URI.create("https://example.com"), builder -> {
+        })) {
+            PutObjectResponse putResponse = client.putObject(
+                PutObjectRequest.builder()
+                    .bucket("test-bucket")
+                    .key("folder/sample.txt")
+                    .contentType("text/plain")
+                    .metadata(Map.of("origin", "integration-test", "purpose", "coverage"))
+                    .tagging("suite=s3&mode=sync")
+                    .build(),
+                RequestBody.fromString("payload-body")
+            );
+
+            ResponseBytes<GetObjectResponse> getResponse = client.getObjectAsBytes(
+                GetObjectRequest.builder()
+                    .bucket("test-bucket")
+                    .key("folder/sample.txt")
+                    .range("bytes=0-4")
+                    .responseContentType("text/plain")
+                    .checksumMode(ChecksumMode.ENABLED)
+                    .build()
+            );
+
+            assertThat(putResponse.eTag()).isEqualTo("\"put-etag\"");
+            assertThat(putResponse.versionId()).isEqualTo("put-version");
+
+            assertThat(getResponse.asUtf8String()).isEqualTo("hello from aws sdk");
+            assertThat(getResponse.response().contentType()).isEqualTo("text/plain");
+            assertThat(getResponse.response().contentLength()).isEqualTo(18L);
+            assertThat(getResponse.response().eTag()).isEqualTo("\"get-etag\"");
+            assertThat(getResponse.response().versionId()).isEqualTo("get-version");
+            assertThat(getResponse.response().metadata()).containsEntry("origin", "integration-test");
+
+            assertThat(httpClient.requests()).hasSize(2);
+
+            RecordedRequest putRequest = httpClient.requests().get(0);
+            assertThat(putRequest.method()).isEqualTo(SdkHttpMethod.PUT);
+            assertThat(putRequest.encodedPath()).isEqualTo("/test-bucket/folder/sample.txt");
+            assertThat(putRequest.headerValue("Content-Type")).hasValue("text/plain");
+            assertThat(putRequest.headerValue("x-amz-meta-origin")).hasValue("integration-test");
+            assertThat(putRequest.headerValue("x-amz-meta-purpose")).hasValue("coverage");
+            assertThat(putRequest.headerValue("x-amz-tagging")).hasValue("suite=s3&mode=sync");
+            assertThat(putRequest.bodyUtf8()).isEqualTo("payload-body");
+
+            RecordedRequest getRequest = httpClient.requests().get(1);
+            assertThat(getRequest.method()).isEqualTo(SdkHttpMethod.GET);
+            assertThat(getRequest.encodedPath()).isEqualTo("/test-bucket/folder/sample.txt");
+            assertThat(getRequest.headerValue("Range")).hasValue("bytes=0-4");
+            assertThat(getRequest.headerValue("x-amz-checksum-mode")).hasValue("ENABLED");
+            assertThat(getRequest.queryValue("response-content-type")).hasValue("text/plain");
+        }
+    }
+
+    @Test
+    void listObjectsV2PaginatorTraversesAllPages() {
+        RecordingHttpClient httpClient = new RecordingHttpClient()
+            .enqueue(MockHttpResponse.xml(200, """
+                <ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <Name>logs-bucket</Name>
+                  <Prefix>logs/</Prefix>
+                  <MaxKeys>1</MaxKeys>
+                  <KeyCount>1</KeyCount>
+                  <IsTruncated>true</IsTruncated>
+                  <NextContinuationToken>token-2</NextContinuationToken>
+                  <Contents>
+                    <Key>logs/app.log</Key>
+                    <LastModified>2024-03-10T12:00:00.000Z</LastModified>
+                    <ETag>"etag-1"</ETag>
+                    <Size>12</Size>
+                    <StorageClass>STANDARD</StorageClass>
+                  </Contents>
+                  <CommonPrefixes>
+                    <Prefix>logs/archive/</Prefix>
+                  </CommonPrefixes>
+                </ListBucketResult>
+                """))
+            .enqueue(MockHttpResponse.xml(200, """
+                <ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <Name>logs-bucket</Name>
+                  <Prefix>logs/</Prefix>
+                  <MaxKeys>1</MaxKeys>
+                  <KeyCount>1</KeyCount>
+                  <IsTruncated>false</IsTruncated>
+                  <Contents>
+                    <Key>logs/archive/app-2.log</Key>
+                    <LastModified>2024-03-10T12:01:00.000Z</LastModified>
+                    <ETag>"etag-2"</ETag>
+                    <Size>18</Size>
+                    <StorageClass>STANDARD</StorageClass>
+                  </Contents>
+                </ListBucketResult>
+                """));
+
+        try (S3Client client = createClient(httpClient, URI.create("https://example.com"), builder -> {
+        })) {
+            ListObjectsV2Request request = ListObjectsV2Request.builder()
+                .bucket("logs-bucket")
+                .prefix("logs/")
+                .delimiter("/")
+                .maxKeys(1)
+                .build();
+
+            List<ListObjectsV2Response> pages = client.listObjectsV2Paginator(request).stream().toList();
+            List<String> keys = pages.stream()
+                .flatMap(page -> page.contents().stream())
+                .map(S3Object::key)
+                .toList();
+            List<String> commonPrefixes = pages.stream()
+                .flatMap(page -> page.commonPrefixes().stream())
+                .map(CommonPrefix::prefix)
+                .toList();
+
+            assertThat(pages).hasSize(2);
+            assertThat(pages.get(0).nextContinuationToken()).isEqualTo("token-2");
+            assertThat(pages.get(1).isTruncated()).isFalse();
+            assertThat(keys).containsExactly("logs/app.log", "logs/archive/app-2.log");
+            assertThat(commonPrefixes).containsExactly("logs/archive/");
+
+            assertThat(httpClient.requests()).hasSize(2);
+
+            RecordedRequest firstRequest = httpClient.requests().get(0);
+            assertThat(firstRequest.method()).isEqualTo(SdkHttpMethod.GET);
+            assertThat(firstRequest.encodedPath()).isEqualTo("/logs-bucket");
+            assertThat(firstRequest.queryValue("list-type")).hasValue("2");
+            assertThat(firstRequest.queryValue("prefix")).hasValue("logs/");
+            assertThat(firstRequest.queryValue("delimiter")).hasValue("/");
+            assertThat(firstRequest.queryValue("max-keys")).hasValue("1");
+            assertThat(firstRequest.queryValue("continuation-token")).isEmpty();
+
+            RecordedRequest secondRequest = httpClient.requests().get(1);
+            assertThat(secondRequest.queryValue("continuation-token")).hasValue("token-2");
+        }
+    }
+
+    @Test
+    void deleteObjectsMarshalsBatchDeleteRequestAndParsesMixedResults() {
+        RecordingHttpClient httpClient = new RecordingHttpClient()
+            .enqueue(MockHttpResponse.xml(200, """
+                <DeleteResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <Deleted>
+                    <Key>docs/quarterly-report.txt</Key>
+                    <VersionId>version-1</VersionId>
+                    <DeleteMarker>true</DeleteMarker>
+                    <DeleteMarkerVersionId>marker-1</DeleteMarkerVersionId>
+                  </Deleted>
+                  <Error>
+                    <Key>docs/protected.txt</Key>
+                    <VersionId>version-2</VersionId>
+                    <Code>AccessDenied</Code>
+                    <Message>Access Denied</Message>
+                  </Error>
+                </DeleteResult>
+                """));
+
+        try (S3Client client = createClient(httpClient, URI.create("https://example.com"), builder -> {
+        })) {
+            DeleteObjectsResponse deleteResponse = client.deleteObjects(request -> request
+                .bucket("docs-bucket")
+                .delete(Delete.builder()
+                    .objects(List.of(
+                        ObjectIdentifier.builder()
+                            .key("docs/quarterly-report.txt")
+                            .versionId("version-1")
+                            .build(),
+                        ObjectIdentifier.builder()
+                            .key("docs/protected.txt")
+                            .versionId("version-2")
+                            .build()
+                    ))
+                    .quiet(true)
+                    .build()));
+
+            assertThat(deleteResponse.deleted()).hasSize(1);
+            DeletedObject deletedObject = deleteResponse.deleted().get(0);
+            assertThat(deletedObject.key()).isEqualTo("docs/quarterly-report.txt");
+            assertThat(deletedObject.versionId()).isEqualTo("version-1");
+            assertThat(deletedObject.deleteMarker()).isTrue();
+            assertThat(deletedObject.deleteMarkerVersionId()).isEqualTo("marker-1");
+
+            assertThat(deleteResponse.errors()).hasSize(1);
+            S3Error error = deleteResponse.errors().get(0);
+            assertThat(error.key()).isEqualTo("docs/protected.txt");
+            assertThat(error.versionId()).isEqualTo("version-2");
+            assertThat(error.code()).isEqualTo("AccessDenied");
+            assertThat(error.message()).isEqualTo("Access Denied");
+
+            assertThat(httpClient.requests()).hasSize(1);
+
+            RecordedRequest deleteRequest = httpClient.requests().get(0);
+            assertThat(deleteRequest.method()).isEqualTo(SdkHttpMethod.POST);
+            assertThat(deleteRequest.encodedPath()).isEqualTo("/docs-bucket");
+            assertThat(deleteRequest.queryParameters().containsKey("delete")).isTrue();
+            assertThat(deleteRequest.headerValue("Content-MD5")).hasValue(md5Base64(deleteRequest.body()));
+            assertThat(deleteRequest.bodyUtf8()).contains("<Delete xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">")
+                .contains("<Quiet>true</Quiet>")
+                .contains("<Key>docs/quarterly-report.txt</Key>")
+                .contains("<VersionId>version-1</VersionId>")
+                .contains("<Key>docs/protected.txt</Key>")
+                .contains("<VersionId>version-2</VersionId>");
+        }
+    }
+
+    @Test
+    void copyObjectMarshalsCopySourceHeadersAndParsesXmlResult() {
+        RecordingHttpClient httpClient = new RecordingHttpClient()
+            .enqueue(MockHttpResponse.of(
+                200,
+                Map.of(
+                    "x-amz-copy-source-version-id", List.of("source-version"),
+                    "x-amz-version-id", List.of("destination-version")
+                ),
+                """
+                <CopyObjectResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <LastModified>2024-03-10T12:34:56.000Z</LastModified>
+                  <ETag>\"copied-etag\"</ETag>
+                </CopyObjectResult>
+                """.getBytes(StandardCharsets.UTF_8)
+            ));
+
+        try (S3Client client = createClient(httpClient, URI.create("https://example.com"), builder -> {
+        })) {
+            CopyObjectResponse copyResponse = client.copyObject(request -> request
+                .sourceBucket("source-bucket")
+                .sourceKey("reports/annual summary.txt")
+                .sourceVersionId("source-version")
+                .destinationBucket("archive-bucket")
+                .destinationKey("archive/annual-summary.txt")
+                .metadataDirective(MetadataDirective.REPLACE)
+                .metadata(Map.of("origin", "integration-test"))
+                .tagging("suite=s3&operation=copy"));
+
+            assertThat(copyResponse.copySourceVersionId()).isEqualTo("source-version");
+            assertThat(copyResponse.versionId()).isEqualTo("destination-version");
+            assertThat(copyResponse.copyObjectResult().eTag()).isEqualTo("\"copied-etag\"");
+            assertThat(copyResponse.copyObjectResult().lastModified()).isEqualTo(Instant.parse("2024-03-10T12:34:56Z"));
+
+            assertThat(httpClient.requests()).hasSize(1);
+
+            RecordedRequest copyRequest = httpClient.requests().get(0);
+            assertThat(copyRequest.method()).isEqualTo(SdkHttpMethod.PUT);
+            assertThat(copyRequest.encodedPath()).isEqualTo("/archive-bucket/archive/annual-summary.txt");
+            assertThat(copyRequest.headerValue("x-amz-copy-source"))
+                .hasValue("source-bucket/reports/annual%20summary.txt?versionId=source-version");
+            assertThat(copyRequest.headerValue("x-amz-metadata-directive")).hasValue("REPLACE");
+            assertThat(copyRequest.headerValue("x-amz-meta-origin")).hasValue("integration-test");
+            assertThat(copyRequest.headerValue("x-amz-tagging")).hasValue("suite=s3&operation=copy");
+            assertThat(copyRequest.body()).isEmpty();
+        }
+    }
+
+    @Test
+    void multipartUploadOperationsMarshalXmlPayloadsAndParseResponses() {
+        RecordingHttpClient httpClient = new RecordingHttpClient()
+            .enqueue(MockHttpResponse.xml(200, """
+                <InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <Bucket>media-bucket</Bucket>
+                  <Key>videos/demo.mp4</Key>
+                  <UploadId>upload-123</UploadId>
+                </InitiateMultipartUploadResult>
+                """))
+            .enqueue(MockHttpResponse.of(
+                200,
+                Map.of("ETag", List.of("\"part-etag\"")),
+                new byte[0]
+            ))
+            .enqueue(MockHttpResponse.xml(200, """
+                <CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                  <Location>https://example.com/media-bucket/videos/demo.mp4</Location>
+                  <Bucket>media-bucket</Bucket>
+                  <Key>videos/demo.mp4</Key>
+                  <ETag>"complete-etag"</ETag>
+                </CompleteMultipartUploadResult>
+                """));
+
+        try (S3Client client = createClient(httpClient, URI.create("https://example.com"), builder -> {
+        })) {
+            CreateMultipartUploadResponse createResponse = client.createMultipartUpload(request -> request
+                .bucket("media-bucket")
+                .key("videos/demo.mp4")
+                .metadata(Map.of("origin", "integration-test"))
+                .tagging("stage=multipart&suite=s3"));
+
+            UploadPartResponse uploadPartResponse = client.uploadPart(request -> request
+                    .bucket("media-bucket")
+                    .key("videos/demo.mp4")
+                    .uploadId(createResponse.uploadId())
+                    .partNumber(1),
+                RequestBody.fromString("first-part"));
+
+            String completedEtag = uploadPartResponse.eTag();
+            client.completeMultipartUpload(request -> request
+                .bucket("media-bucket")
+                .key("videos/demo.mp4")
+                .uploadId(createResponse.uploadId())
+                .multipartUpload(CompletedMultipartUpload.builder()
+                    .parts(CompletedPart.builder()
+                        .partNumber(1)
+                        .eTag(completedEtag)
+                        .build())
+                    .build()));
+
+            assertThat(createResponse.bucket()).isEqualTo("media-bucket");
+            assertThat(createResponse.key()).isEqualTo("videos/demo.mp4");
+            assertThat(createResponse.uploadId()).isEqualTo("upload-123");
+            assertThat(uploadPartResponse.eTag()).isEqualTo("\"part-etag\"");
+
+            assertThat(httpClient.requests()).hasSize(3);
+
+            RecordedRequest createRequest = httpClient.requests().get(0);
+            assertThat(createRequest.method()).isEqualTo(SdkHttpMethod.POST);
+            assertThat(createRequest.encodedPath()).isEqualTo("/media-bucket/videos/demo.mp4");
+            assertThat(createRequest.queryParameters().containsKey("uploads")).isTrue();
+            assertThat(createRequest.headerValue("x-amz-meta-origin")).hasValue("integration-test");
+            assertThat(createRequest.headerValue("x-amz-tagging")).hasValue("stage=multipart&suite=s3");
+            assertThat(createRequest.body()).isEmpty();
+
+            RecordedRequest uploadRequest = httpClient.requests().get(1);
+            assertThat(uploadRequest.method()).isEqualTo(SdkHttpMethod.PUT);
+            assertThat(uploadRequest.queryValue("uploadId")).hasValue("upload-123");
+            assertThat(uploadRequest.queryValue("partNumber")).hasValue("1");
+            assertThat(uploadRequest.bodyUtf8()).isEqualTo("first-part");
+
+            RecordedRequest completeRequest = httpClient.requests().get(2);
+            assertThat(completeRequest.method()).isEqualTo(SdkHttpMethod.POST);
+            assertThat(completeRequest.queryValue("uploadId")).hasValue("upload-123");
+            assertThat(completeRequest.bodyUtf8()).contains("<CompleteMultipartUpload")
+                .contains("<PartNumber>1</PartNumber>")
+                .contains("<ETag>&quot;part-etag&quot;</ETag>");
+        }
+    }
+
+    private static S3Client createClient(
+        RecordingHttpClient httpClient,
+        URI endpoint,
+        Consumer<S3Configuration.Builder> configurationCustomizer
+    ) {
+        S3Configuration.Builder configurationBuilder = S3Configuration.builder()
+            .pathStyleAccessEnabled(true)
+            .chunkedEncodingEnabled(false)
+            .checksumValidationEnabled(false);
+        configurationCustomizer.accept(configurationBuilder);
+
+        return S3Client.builder()
+            .region(Region.US_WEST_2)
+            .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("access-key", "secret-key")))
+            .endpointOverride(endpoint)
+            .serviceConfiguration(configurationBuilder.build())
+            .httpClient(httpClient)
+            .build();
+    }
+
+    private record MockHttpResponse(int statusCode, Map<String, List<String>> headers, byte[] body) {
+        private static MockHttpResponse of(int statusCode, Map<String, List<String>> headers, byte[] body) {
+            return new MockHttpResponse(statusCode, copyMultiValueMap(headers), body.clone());
+        }
+
+        private static MockHttpResponse xml(int statusCode, String body) {
+            return new MockHttpResponse(
+                statusCode,
+                Map.of("Content-Type", List.of("application/xml")),
+                body.getBytes(StandardCharsets.UTF_8)
+            );
+        }
+    }
+
+    private record RecordedRequest(
+        SdkHttpMethod method,
+        String host,
+        String encodedPath,
+        Map<String, List<String>> queryParameters,
+        Map<String, List<String>> headers,
+        byte[] body
+    ) {
+        private Optional<String> headerValue(String name) {
+            return findFirstValue(headers, name);
+        }
+
+        private Optional<String> queryValue(String name) {
+            return findFirstValue(queryParameters, name);
+        }
+
+        private String bodyUtf8() {
+            return new String(body, StandardCharsets.UTF_8);
+        }
+
+        private static RecordedRequest from(SdkHttpRequest request, byte[] body) {
+            return new RecordedRequest(
+                request.method(),
+                request.host(),
+                request.encodedPath(),
+                copyMultiValueMap(request.rawQueryParameters()),
+                copyMultiValueMap(request.headers()),
+                body.clone()
+            );
+        }
+    }
+
+    private static final class RecordingHttpClient implements SdkHttpClient {
+        private final Deque<MockHttpResponse> queuedResponses = new ArrayDeque<>();
+        private final List<RecordedRequest> requests = new ArrayList<>();
+
+        private RecordingHttpClient enqueue(MockHttpResponse response) {
+            queuedResponses.addLast(response);
+            return this;
+        }
+
+        private List<RecordedRequest> requests() {
+            return requests;
+        }
+
+        @Override
+        public ExecutableHttpRequest prepareRequest(HttpExecuteRequest request) {
+            return new ExecutableHttpRequest() {
+                @Override
+                public HttpExecuteResponse call() {
+                    byte[] requestBody = request.contentStreamProvider()
+                        .map(Content -> readAllBytes(Content.newStream()))
+                        .orElseGet(() -> new byte[0]);
+                    requests.add(RecordedRequest.from(request.httpRequest(), requestBody));
+
+                    MockHttpResponse response = queuedResponses.pollFirst();
+                    if (response == null) {
+                        throw new IllegalStateException("No mock response queued for " + request.httpRequest().getUri());
+                    }
+
+                    SdkHttpFullResponse.Builder responseBuilder = SdkHttpFullResponse.builder()
+                        .statusCode(response.statusCode())
+                        .content(AbortableInputStream.create(new ByteArrayInputStream(response.body())));
+                    response.headers().forEach(responseBuilder::putHeader);
+
+                    return HttpExecuteResponse.builder()
+                        .response(responseBuilder.build())
+                        .responseBody(AbortableInputStream.create(new ByteArrayInputStream(response.body())))
+                        .build();
+                }
+
+                @Override
+                public void abort() {
+                }
+            };
+        }
+
+        @Override
+        public String clientName() {
+            return "recording-http-client";
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    private static Map<String, List<String>> copyMultiValueMap(Map<String, List<String>> source) {
+        Map<String, List<String>> copy = new LinkedHashMap<>();
+        source.forEach((key, value) -> copy.put(key, value == null ? null : new ArrayList<>(value)));
+        return copy;
+    }
+
+    private static Optional<String> findFirstValue(Map<String, List<String>> values, String name) {
+        return values.entrySet().stream()
+            .filter(entry -> entry.getKey().equalsIgnoreCase(name))
+            .map(Map.Entry::getValue)
+            .filter(value -> value != null && !value.isEmpty())
+            .map(value -> value.get(0))
+            .findFirst();
+    }
+
+    private static String md5Base64(byte[] content) {
+        try {
+            return Base64.getEncoder().encodeToString(MessageDigest.getInstance("MD5").digest(content));
+        } catch (NoSuchAlgorithmException exception) {
+            throw new IllegalStateException("MD5 digest is unavailable", exception);
+        }
+    }
+
+    private static byte[] readAllBytes(InputStream inputStream) {
+        try (InputStream closableInputStream = inputStream) {
+            return closableInputStream.readAllBytes();
+        } catch (IOException exception) {
+            throw new UncheckedIOException(exception);
+        }
+    }
+}

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/src/test/java/software_amazon_awssdk/s3/S3Test.java
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/src/test/java/software_amazon_awssdk/s3/S3Test.java
@@ -44,6 +44,7 @@ import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.s3.S3Client;
 import software.amazon.awssdk.services.s3.S3Configuration;
 import software.amazon.awssdk.services.s3.S3Uri;
+import software.amazon.awssdk.services.s3.model.ChecksumAlgorithm;
 import software.amazon.awssdk.services.s3.model.ChecksumMode;
 import software.amazon.awssdk.services.s3.model.CommonPrefix;
 import software.amazon.awssdk.services.s3.model.CompletedMultipartUpload;
@@ -66,14 +67,13 @@ import software.amazon.awssdk.services.s3.model.S3Error;
 import software.amazon.awssdk.services.s3.model.S3Object;
 import software.amazon.awssdk.services.s3.model.UploadPartResponse;
 
-class S3Test {
+public class S3Test {
 
     @Test
     void configurationAndUtilitiesSupportPathStyleUrlsAndUriParsing() {
         S3Configuration configuration = S3Configuration.builder()
             .pathStyleAccessEnabled(true)
             .chunkedEncodingEnabled(false)
-            .checksumValidationEnabled(false)
             .multiRegionEnabled(true)
             .build();
 
@@ -83,7 +83,6 @@ class S3Test {
 
         assertThat(configuration.pathStyleAccessEnabled()).isTrue();
         assertThat(configuration.chunkedEncodingEnabled()).isFalse();
-        assertThat(configuration.checksumValidationEnabled()).isFalse();
         assertThat(configuration.multiRegionEnabled()).isTrue();
         assertThat(copiedConfiguration.multiRegionEnabled()).isFalse();
 
@@ -303,6 +302,7 @@ class S3Test {
         })) {
             DeleteObjectsResponse deleteResponse = client.deleteObjects(request -> request
                 .bucket("docs-bucket")
+                .checksumAlgorithm(ChecksumAlgorithm.SHA256)
                 .delete(Delete.builder()
                     .objects(List.of(
                         ObjectIdentifier.builder()
@@ -337,7 +337,8 @@ class S3Test {
             assertThat(deleteRequest.method()).isEqualTo(SdkHttpMethod.POST);
             assertThat(deleteRequest.encodedPath()).isEqualTo("/docs-bucket");
             assertThat(deleteRequest.queryParameters().containsKey("delete")).isTrue();
-            assertThat(deleteRequest.headerValue("Content-MD5")).hasValue(md5Base64(deleteRequest.body()));
+            assertThat(deleteRequest.headerValue("x-amz-sdk-checksum-algorithm")).hasValue("SHA256");
+            assertThat(deleteRequest.headerValue("x-amz-checksum-sha256")).hasValue(sha256Base64(deleteRequest.body()));
             assertThat(deleteRequest.bodyUtf8()).contains("<Delete xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">")
                 .contains("<Quiet>true</Quiet>")
                 .contains("<Key>docs/quarterly-report.txt</Key>")
@@ -483,8 +484,7 @@ class S3Test {
     ) {
         S3Configuration.Builder configurationBuilder = S3Configuration.builder()
             .pathStyleAccessEnabled(true)
-            .chunkedEncodingEnabled(false)
-            .checksumValidationEnabled(false);
+            .chunkedEncodingEnabled(false);
         configurationCustomizer.accept(configurationBuilder);
 
         return S3Client.builder()
@@ -612,11 +612,11 @@ class S3Test {
             .findFirst();
     }
 
-    private static String md5Base64(byte[] content) {
+    private static String sha256Base64(byte[] content) {
         try {
-            return Base64.getEncoder().encodeToString(MessageDigest.getInstance("MD5").digest(content));
+            return Base64.getEncoder().encodeToString(MessageDigest.getInstance("SHA-256").digest(content));
         } catch (NoSuchAlgorithmException exception) {
-            throw new IllegalStateException("MD5 digest is unavailable", exception);
+            throw new IllegalStateException("SHA-256 digest is unavailable", exception);
         }
     }
 

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/user-code-filter.json
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "software.amazon.awssdk.**"
+    }
+  ]
+}

--- a/tests/src/software.amazon.awssdk/s3/2.30.0/user-code-filter.json
+++ b/tests/src/software.amazon.awssdk/s3/2.30.0/user-code-filter.json
@@ -4,7 +4,10 @@
       "excludeClasses" : "**"
     },
     {
-      "includeClasses" : "software.amazon.awssdk.**"
+      "includeClasses" : "software.amazon.awssdk.services.s3.**"
+    },
+    {
+      "includeClasses" : "software_amazon_awssdk.s3.**"
     }
   ]
 }

